element-connector/src/main/java/org/eclipse/californium/elements/RawData.java

```diff
@@ -225,13 +227,15 @@ public static RawData inbound(byte[] data, InetSocketAddress address, Principal
 	 * 
 	 * @param data the data to send.
 	 * @param address the IP address and port the data is to be sent to.
+	 * @param correlationContext correlation context for sending data (may be <code>null</code>).
 	 * @param callback the handler to call when this message has been sent (may be <code>null</code>).
 	 * @param useMulticast indicates whether the data should be sent using a multicast message.
 	 * @return the raw data object containing the outbound message.
 	 * @throws NullPointerException if data or address is <code>null</code>.
 	 */
-	public static RawData outbound(byte[] data, InetSocketAddress address, MessageCallback callback, boolean useMulticast) {
+	public static RawData outbound(byte[] data, InetSocketAddress address, CorrelationContext correlationContext, MessageCallback callback, boolean useMulticast) {
```

Comment:
The provided CorrelationContext may not contain all informations, which the connector can provide. E.g. someone may decide to put the pask indentity in and provide a matcher, which checks, if the dtls session actually used to encrypt the data matches that identity.
In that case, either the additional values must be added or a new correlation context must be handled back by the callback. I decided to use the already used one.